geoportal/c2cgeoportal_geoportal/views/layers.py

```diff
@@ -469,22 +469,32 @@ def query_enumerate_attribute_values(dbsession, layerinfos, fieldname):
         return dbsession.query(distinct(attribute)).order_by(attribute).all()
 
 
-def get_layer_class(layer, with_exclude=False):
-    if with_exclude:
-        # Exclude the columns used to record the last features update
-        exclude = [] if layer.exclude_properties is None else layer.exclude_properties.split(",")
+def get_layer_class(layer, for_metadata=False):
```

Comment:
`with_last_update_columns=True` might make more sense here as it is also for LiguaExtractor, not only for metadata view.

Note that moving editingAttributesOrder and readonlyAttributes inside if block will create two distinct classes in cache for metadata and protocol, even if there is no last_update_columns, but this might not have a big impact on memory usage, unless there is thousands of editable layers.

Reply:
Your right I will get them out for the cache ...

Reply:
Done :-)